src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -236,21 +239,40 @@ def add_resolved_requirements(self, reqs, platforms=None):
     :param log: Use this logger.
     :param platforms: A list of :class:`Platform`s to resolve requirements for.
                       Defaults to the platforms specified by PythonSetup.
+    :param bool override_ptex_skip: When this PexBuilderWrapper is configured with
+                                    generate_ptex=True, this method won't add any distributions to
+                                    the output pex. The internal implementation of this class adds a
+                                    pex dependency to the output ptex file, and therefore needs to
+                                    override the default behavior of this method.
     """
-    distributions = self._resolve_distributions_by_platform(reqs, platforms=platforms)
-    locations = set()
+    distributions, transitive_requirements = self._resolve_distributions_by_platform(reqs, platforms=platforms)
+    locations: Set[str] = set()
     for platform, dists in distributions.items():
       for dist in dists:
         if dist.location not in locations:
-          self._log.debug(f'  Dumping distribution: .../{os.path.basename(dist.location)}')
-          self.add_distribution(dist)
+          if self._generate_ptex and not override_ptex_skip:
+            self._log.debug(f'  *AVOIDING* dumping distribution into ptex: .../{os.path.basename(dist.location)}')
+          else:
+            self._log.debug(f'  Dumping distribution: .../{os.path.basename(dist.location)}')
+            self.add_distribution(dist)
         locations.add(dist.location)
-
-  def _resolve_multi(self, interpreter, requirements, platforms, find_links):
+    # In addition to the top-level requirements, we add all the requirements matching the resolved
+    # distributions to the resulting pex. If `generate_ptex=True` is set, we need to have all the
+    # transitive requirements resolved in order to hydrate the .ptex with an intransitive resolve.
+    self.add_direct_requirements(transitive_requirements)
```

Comment:
Should this only be done conditionally if `override_ptex_skip` is set? We have the parameter anyway, so...

Reply:
I made it use the same conditional as above in https://github.com/pantsbuild/pants/pull/8793/commits/b132ad0de88cf52e37e3e67c79b51ae8a6110f17 -- is that what you're referring to?